### PR TITLE
Fix white flash loading dark-theme pages

### PR DESCRIPTION
**Pull request by lann**
Move the `dark-theme` class from `<body>` to `<html>` and move the related JS into `<head>` so it will run before first page draw.

re: https://github.com/fermyon/spin/issues/253

### CODE REVIEW

File: docs/layout-docs.html [hunk 1]
```diff
@@ -35,7 +35,51 @@
     />
     <link href="https://fonts.googleapis.com/css2?family=Sen:wght@400;700&amp;display=swap" rel="stylesheet" />
     <link rel="stylesheet" href="../scss/fermyon.scss" />
-
+    <script>
```
Comment (radu-matei) [inline docs/layout-docs.html:38]: Do we want to pull this script into a file rather than duplicating it in all layouts?

**Review by radu-matei — APPROVED** [review]

File: docs/layout-docs.html (hunk 1 above)
Comment (lann) [reply docs/layout-docs.html:38]: Scripts in `<head>` block first render until they are loaded so there is some marginal benefit in keeping them inline. Ideally they'd be inlined by your build tooling but I think thats a question for @flynnduism

Comment (radu-matei) [reply docs/layout-docs.html:38]: Ah, thanks for the explanation.
I think we can merge and explore whether we want to somehow use tooling to inline them later.

**Review by flynnduism — APPROVED** [review]
Thanks @lann this works great and addresses the flash issue. Much smoother!!

**Pull request merged**